src/main/java/net/minecraftforge/items/ItemStackHandler.java

```diff
@@ -162,6 +172,12 @@ public boolean isItemValid(int slot, @NotNull ItemStack stack)
         return true;
     }
 
+    @Override
+    public IOGuarantees getIOGuarantees()
```

Comment:
I wonder if some implementations might override methods in a way that make the handler not strict anymore. However I don't think we should do anything about it cause it wouldn't be practical.

Reply:
I think being an item stack handler imposes some contracts already on implementors, so I don't think we really need to worry about it.